internal/pkg/service/stream/storage/repository/volume/repo_test.go

```diff
@@ -0,0 +1,170 @@
+	"github.com/stretchr/testify/require"
+	"go.etcd.io/etcd/client/v3/concurrency"
+
+	deps "github.com/keboola/keboola-as-code/internal/pkg/service/common/dependencies"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/utctime"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/dependencies"
+	volume "github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/level/local/volume/model"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/etcdhelper"
+)
+
+func TestRepository_Volume(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+
+	clk := clock.NewMock()
+	clk.Set(utctime.MustParse("2000-01-03T01:00:00.000Z").Time())
+
+	// Get services
+	d, mocked := dependencies.NewMockedLocalStorageScope(t, deps.WithClock(clk))
+	client := mocked.TestEtcdClient()
+	r := d.StorageRepository().Volume()
+
+	// Create session
+	session, err := concurrency.NewSession(client)
+	require.NoError(t, err)
+
+	// Fixtures
+	volume1 := volume.Metadata{
+		ID: volume.ID("my-volume-1"),
+		Spec: volume.Spec{
+			NodeID: "writer-node",
+			Path:   "hdd/001",
+			Type:   "hdd",
+			Label:  "001",
+		},
+	}
+	volume2 := volume.Metadata{
+		ID: volume.ID("my-volume-2"),
+		Spec: volume.Spec{
+			NodeID: "writer-node",
+			Path:   "hdd/002",
+			Type:   "hdd",
+			Label:  "002",
+		},
+	}
+	volume3 := volume.Metadata{
+		ID: volume.ID("my-volume-3"),
+		Spec: volume.Spec{
+			NodeID: "reader-node",
+			Path:   "hdd/003",
+			Type:   "hdd",
+			Label:  "003",
+		},
+	}
+	volume4 := volume.Metadata{
+		ID: volume.ID("my-volume-4"),
+		Spec: volume.Spec{
+			NodeID: "reader-node",
+			Path:   "hdd/004",
+			Type:   "hdd",
+			Label:  "004",
+		},
+	}
+
+	// Empty
+	// -----------------------------------------------------------------------------------------------------------------
+	{
+		// Empty list
+		result, err := r.ListWriterVolumes().Do(ctx).AllKVs()
+		require.NoError(t, err)
+		assert.Empty(t, result)
+
+		result, err = r.ListReaderVolumes().Do(ctx).AllKVs()
+		require.NoError(t, err)
+		assert.Empty(t, result)
+	}
+
+	// Registration
+	// -----------------------------------------------------------------------------------------------------------------
+	{
+		// Register
+		require.NoError(t, r.RegisterWriterVolume(volume1, session.Lease()).Do(ctx).Err())
+		require.NoError(t, r.RegisterWriterVolume(volume2, session.Lease()).Do(ctx).Err())
+
+		require.NoError(t, r.RegisterReaderVolume(volume3, session.Lease()).Do(ctx).Err())
+		require.NoError(t, r.RegisterReaderVolume(volume4, session.Lease()).Do(ctx).Err())
+	}
+	{
+		// List
+		result, err := r.ListWriterVolumes().Do(ctx).All()
+		require.NoError(t, err)
+		assert.Equal(t, []volume.Metadata{volume1, volume2}, result)
+
+		result, err = r.ListReaderVolumes().Do(ctx).All()
+		require.NoError(t, err)
+		assert.Equal(t, []volume.Metadata{volume3, volume4}, result)
+	}
+	{
+		// Check etcd state
+		etcdhelper.AssertKVsString(t, client, `
```

Comment:
Why dont you use comparison with file as in previous PR?

Reply:
Thanks, typo, I will fix it.